### PR TITLE
Docs: Added Experimental Edit pattern

### PR DESCRIPTION
**Pull request by chloepoulterdesign**
## Closes:
https://github.ibm.com/CloudIntegrationDesign/DesignSystemAdoptionGuild/issues/85
https://github.ibm.com/CDAI-design/CDAI-design-website/pull/7

This contributes the CDAI Edit pattern to Carbon. 

### New
- Added the "Edit-pattern" folder in the "Experimental" section
- Added 12 new images in the "Experimental/Images" folder
- Added text to index.mdx
- Added link into Navigation using nav-items.yaml

**Changed**

- nothing changed

**Removed**

- Nothing removed
- Could supersede the edit in Carbon Patterns/Common Actions/Progressive actions with more development

### CODE REVIEW

**Comment by chloepoulterdesign** [comment]
@connor-leech i'd really appreciate your review on this - @solly2 mentioned he thinks it comes across a bit more prescriptive than it really needs to.

**Comment by connor-leech** [comment]
@chloepoulterdesign I don't think it's too prescriptive. In general, being prescriptive with patterns encourages disagreement, which will bolster the pattern in the long run.

This PR looks great.

File: src/pages/experimental/edit-pattern/index.mdx [hunk 1]
```diff
@@ -0,0 +1,201 @@
+---
+title: Edit & Update Pattern
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]: If I was looking for Update pattern guidance, I think I'd probably click on Edit. I think we can safely call it the "Edit pattern" and call out Update in the introduction you already do. It'll still appear upon search if a user can't find it.

File: src/pages/experimental/edit-pattern/index.mdx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+---
+
+<PageDescription>
+
+_Editing_ is making changes to a resource that already exists in the system.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
Editing is making changes to a resource that already exists in the system.
```

**Comment by chloepoulterdesign** [comment]
Sorry @connor-leech i accidentally re-requested your review 😂
If you're all good with it, hopefully @vikkipaterson can also review and approve

File: src/pages/experimental/edit-pattern/index.mdx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+#### Status:
+
+[Experimental](/experimental/overview/)
+
+#### Maintainers:
+
+[Vikki Paterson](https://github.com/vikkipaterson), [Chloe Poulter](https://github.com/chloepoulterdesign)
+
+### Options:
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
### Options:
```
We can't have headers above the anchorlinks here. As you mention that the one should consider the consequences before allowing a deletion, I think we can just delete this header/line below.

File: src/pages/experimental/edit-pattern/index.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+Consider the consequences of editing before allowing it:
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]: Same as comment above.

File: src/pages/experimental/edit-pattern/index.mdx (hunk 3 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/edit-pattern/index.mdx]: So just remove this line?

**Comment by chloepoulterdesign** [comment]
Committed fixes to Connor's comments

File: src/pages/experimental/edit-pattern/index.mdx [hunk 5]
```diff
@@ -0,0 +1,197 @@
+---
+title: Edit Pattern
+---
+
+<PageDescription>
+
+Editing is making changes to a resource that already exists in the system.
+Updating and editing should use the same behaviour as when the resource is created. This should only be varied in situations where the number of editable fields is significantly smaller or larger than during creation, or the context of current settings is required.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
Updating and editing should use the same behavior as when the resource is created. This should only vary in situations where the number of editable fields is significantly smaller or larger than during creation, or the context of current settings is required.
```
This isn't wrong, just switching to American spelling 😝

File: src/pages/experimental/edit-pattern/index.mdx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+Warn the user when the consequence of editing could damage resources or connections in their system.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]: According to our design team, we can't have line breaks in page descriptions. As we address consequences and related actions on each variant, we can probably delete this line. What do you think?

File: src/pages/experimental/edit-pattern/index.mdx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+[Experimental](/experimental/overview/)
+
+#### Maintainers:
+
+[Vikki Paterson](https://github.com/vikkipaterson), [Chloe Poulter](https://github.com/chloepoulterdesign)
+
+<AnchorLinks>
+
+  <AnchorLink>Low impact edit</AnchorLink>
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
  <AnchorLink>Low-impact edit</AnchorLink>
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+  <AnchorLink>High impact edit</AnchorLink>
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
  <AnchorLink>High-impact edit</AnchorLink>
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+  <AnchorLink>Optional extras</AnchorLink>
+
+</AnchorLinks>
+
+![Contextual image of an edit modal](images/01.png)
+
+<Caption>Example of an edit modal in context</Caption>
+
+## Low impact edit
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
## Low-impact edit
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+<Column colLg={8}>
+
+![Example of a low impact edit action in a side panel](images/04.png)
+
+<Caption>Example of a low impact edit action in a side panel</Caption>
+
+</Column>
+</Row>
+
+## High impact edit
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
## High-impact edit
```

File: src/pages/experimental/edit-pattern/index.mdx (hunk 6 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/edit-pattern/index.mdx]: Yes, I think we can cope with that

File: src/pages/experimental/edit-pattern/index.mdx [hunk 11]
```diff
@@ -0,0 +1,197 @@
+---
+title: Edit Pattern
+---
+
+<PageDescription>
+
+Editing is making changes to a resource that already exists in the system.
+Updating and editing should use the same behavior as when the resource is created. This should only vary in situations where the number of editable fields is significantly smaller or larger than during creation, or the context of current settings is required.
+
+Warn the user when the consequence of editing could damage resources or connections in their system.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+</Row>
+
+### Side-panel
+
+If an object was created in a side panel, or there are too many editable fields for a modal, a slide-over panel can be used.
+
+<Row>
+<Column colLg={8}>
+
+![Example of a high impact edit action in a side-panel](images/06.png)
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
![Example of a high-impact edit action in a side-panel](images/06.png)
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+
+<Caption>Example of a high impact edit action in a side-panel</Caption>
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
<Caption>Example of a high-impact edit action in a side-panel</Caption>
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+
+</Column>
+</Row>
+
+High impact side panel edits should also warn the user of destructive edits with a warning before changes are saved.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
High-impact side panel edits should also warn the user of destructive edits with a warning before changes are saved.
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+![Example of a warning screen in a side panel before a user saves changes](images/07.png)
+
+<Caption>
+  Example of a warning screen in a side panel before a user saves changes
+</Caption>
+
+</Column>
+</Row>
+
+### Full-page
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
### Full page
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+<Caption>Example of a single non-editable field</Caption>
+
+</Column>
+</Row>
+
+Where a majority of variables are not editable, consider using an option that will display only editable fields. In a small number of cases, make individual variables editable in modals.
+
+### Success notifications
+
+An optional success notification can be used to confirm that the users' changes have been saved.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
An optional success notification can be used to confirm that the user's changes have been saved.
```

File: src/pages/experimental/edit-pattern/index.mdx (hunk 13 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/edit-pattern/index.mdx]: I thought you'd inserted the hyphen into `side panel` in other reviews, but could be entirely mistaken

File: src/data/nav-items.yaml [hunk 17]
```diff
@@ -134,6 +134,10 @@
       path: /experimental/chatbot/overview
     - title: Create flows
       path: /experimental/create-flows
+    - title: Delete pattern
```
Comment (connor-leech) [inline src/data/nav-items.yaml]:
```suggestion
```
We no longer have a delete pattern, it was merged with Remove, right?

Comment (chloepoulterdesign) [reply src/data/nav-items.yaml]: oh yes - well spotted

File: src/pages/experimental/edit-pattern/index.mdx [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+### Success notifications
+
+An optional success notification can be used to confirm that the user's changes have been saved.
+
+<Row>
+<Column colLg={8}>
+
+![Example of a successful editing toaster notification](images/12.png)
+
+<Caption>Example of a successful editing toaster notification</Caption>
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
<Caption>Example of a successful editing toast notification</Caption>
```
I assume this is supposed to say toast 😂

File: src/pages/experimental/edit-pattern/index.mdx [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+
+</AnchorLinks>
+
+![Contextual image of an edit modal](images/01.png)
+
+<Caption>Example of an edit modal in context</Caption>
+
+## Low-impact edit
+
+Allow editing with no warning. Typically used when editing doesn’t have a wider impact.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
For low-impact edits, allow editing with no warning. This approach is typically used when editing doesn’t have a wider impact.
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+
+<Caption>Example of an edit button in a table</Caption>
+
+</Column>
+</Row>
+
+### Modal
+
+If an object was created in a modal, its editable fields can be made available in a modal. A modal can also be used where a small subset of fields are editable.
+If enough fields are editable to require scrolling, use a side-panel or full-page edit dialog instead.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]: Did you want this to be two separate paragraphs? It looks like there is a line break, but no empty line between them so it'll render as its one paragraph.

File: src/pages/experimental/edit-pattern/index.mdx [hunk 21]
```diff
@@ -0,0 +1,196 @@
...
+![Example of a low consequence edit action in a modal](images/03.png)
+
+<Caption>Example of a low conssequence edit action in a modal</Caption>
+
+</Column>
+</Row>
+
+### Side-panel
+
+If an object was created in a side panel, its editable fields can be made available in a side-panel. A side-panel can also be used where context in the main view of the screen is useful in the editing flow.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
If an object was created in a side panel, its editable fields can be made available in a side panel. A side panel can also be used where context in the main view of the screen is useful in the editing flow.
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 22]
```diff
@@ -0,0 +1,196 @@
...
+
+<Row>
+<Column colLg={8}>
+
+![Example of a low impact edit action in a side panel](images/04.png)
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
![Example of a low-impact edit action in a side panel](images/04.png)
```

File: src/pages/experimental/edit-pattern/index.mdx (hunk 20 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/edit-pattern/index.mdx]: One paragraph is probably fine - I imagine it's the ol' trigger happy thumbs again

File: src/pages/experimental/edit-pattern/index.mdx [hunk 23]
```diff
@@ -0,0 +1,196 @@
...
+
+If an object was created in a modal, its editable fields can be made available in a modal. A modal can also be used where a small subset of fields are editable.
+If enough fields are editable to require scrolling, use a side-panel or full-page edit dialog instead.
+
+<Row>
+<Column colLg={8}>
+
+![Example of a low consequence edit action in a modal](images/03.png)
+
+<Caption>Example of a low conssequence edit action in a modal</Caption>
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
<Caption>Example of a low consequence edit action in a modal</Caption>
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 24]
```diff
@@ -0,0 +1,196 @@
...
+![Example of a low impact edit action in a side panel](images/04.png)
+
+<Caption>Example of a low impact edit action in a side panel</Caption>
+
+</Column>
+</Row>
+
+## High-impact edit
+
+Warn the user of the consequences of editing. Typically used when editing has a wider impact.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
Warn the user of the consequences of editing. This pattern is typically used when editing has a wider impact.
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 25]
```diff
@@ -0,0 +1,196 @@
...
+Inform users of the likely consequences of the edit to their system, and inform them that editing cannot be undone. Include a warning stage.
+
+### Modal
+
+A high impact edit action of only a few fields can use a Danger modal. A second confirm edit modal should also be inserted before the changes are saved. If desired, the confirmation modal can include a [summary of changes](#summary-of-changes).
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
A high-impact edit action of only a few fields can use a Danger modal. A second confirm edit modal should also be inserted before the changes are saved. If desired, the confirmation modal can include a [summary of changes](#summary-of-changes).
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 26]
```diff
@@ -0,0 +1,196 @@
...
+</Caption>
+
+</Column>
+</Row>
+
+## Optional extras
+
+### Summary of changes
+
+Adding a summary of changes to the warning stage of a High-impact flow can help the user manage and understand all of the changes they have made.
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
Adding a summary of changes to the warning stage of a high-impact flow can help the user manage and understand all of the changes they have made.
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 27]
```diff
@@ -0,0 +1,196 @@
...
+</Row>
+
+## High-impact edit
+
+Warn the user of the consequences of editing. This pattern is typically used when editing has a wider impact.
+Inform users of the likely consequences of the edit to their system, and inform them that editing cannot be undone. Include a warning stage.
+
+### Modal
+
+A high-impact edit action of only a few fields can use a Danger modal. A second confirm edit modal should also be inserted before the changes are saved. If desired, the confirmation modal can include a [summary of changes](#summary-of-changes).
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
A high-impact edit action of only a few fields can use a [danger modal]. A second confirm edit modal should also be inserted before the changes are saved. If desired, the confirmation modal can include a [summary of changes](#summary-of-changes).
```

File: src/pages/experimental/edit-pattern/index.mdx [hunk 28]
```diff
@@ -0,0 +1,196 @@
...
+Where a majority of variables are not editable, consider using an option that will display only editable fields. In a small number of cases, make individual variables editable in modals.
+
+### Success notifications
+
+An optional success notification can be used to confirm that the user's changes have been saved.
+
+<Row>
+<Column colLg={8}>
+
+![Example of a successful editing toaster notification](images/12.png)
```
Comment (connor-leech) [inline src/pages/experimental/edit-pattern/index.mdx]:
```suggestion
![Example of a successful editing toast notification](images/12.png)
```

**Pull request merged**